### PR TITLE
Move away from JSR-305 Nullable and Nonnull annotations

### LINKED ISSUE #1633: Move off JSR-305 annotations
See https://github.com/spotbugs/spotbugs/issues/421

The `JSR-305` from findbugs is not longer supported, has split package problems for Java 9+ and may represent an Oracle license violation.

We currently use `javax` annotations for `Nullable` and `Nonnull` we need to eliminate the use of these annotations directly but still support them via annotation mapping.

The following screenshot shows the currently supported annotations by IntelliJ out of the box:

Screenshot 2019-05-07 at 08 30 25

Google tools are moving to the Checker Framework so maybe we should use that since it is supported out of the box by IntelliJ

The other option is our own set of annotations, however that would require end users to configure IntelliJ (and any other tooling) explicitly.

### PR DESCRIPTION
**Pull request by graemerocher**
Fixes #1633

### CODE REVIEW

File: src/main/docs/guide/appendix/breaks.adoc [hunk 1]
```diff
@@ -6,6 +6,14 @@ In Micronaut 1.x the ann:context.annotation.Executable[] annotation was a meta-a
 
 In Micronaut 2.x and above this is no longer the case and if you need an api:inject.ExecutableMethod[] generated you should add the ann:context.annotation.Executable[] annotation to any custom AOP advice you specify otherwise the method will not be available via the api:inject.BeanDefinition[] interface (using for example the api:BeanDefinition.getExecutableMethods[] method).
 
+=== Spot Bugs Instead of JSR-305 Nullable/NonNull Annotations
```
Comment (jameskleeh) [inline src/main/docs/guide/appendix/breaks.adoc]: I would put this under whats new and just add a blurb here about how the dependency isn't part of the API of core any longer

**Comment by dave08** [comment]
This won't cause any problems with Intellij seeing nullable or non-nullables for Kotlin?

**Comment by graemerocher** [comment]
No IntelliJ supports this the same

**Comment by andresmargalef** [comment]
@graemerocher what is the motivation for this change, seems interesting to know a little more :D

**Comment by graemerocher** [comment]
@andresmargalef The reasons are summed up in this comment https://github.com/google/guava/issues/2960#issue-263628666

**Pull request merged**